File: HR_Management/ClientApp/HrManagement/src/app/models/Leave/leave-request-vm.ts
export type LeaveRequestStatus = 'Pending' | 'Approved' | 'Rejected';
export class LeaveRequestVM {
  id: number = 0;
  employeeId: number = 0;
  leaveTypeId: number = 0;
  startDate: string | null = null;
  endDate: string | null = null;
  reason: string | null = null;
  status: LeaveRequestStatus = 'Pending';
  employeeName: string | null = null;
  leaveTypeName: string | null = null;
  createdAt:string|null=null

}

File: HR_Management.Core/DTOs/WebApplication/LeaveRequest/LeaveRequestVM.cs
using HR_Management.Core.Entities.Leaves;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HR_Management.Core.DTOs.WebApplication.LeaveRequest
{
    public class LeaveRequestVM
    {
        public int Id { get; set; }
        public string EmployeeName { get; set; } = string.Empty;
        public string LeaveTypeName { get; set; } = string.Empty;
        public string StartDate { get; set; }
        public string EndDate { get; set; }
        public int LeaveDuration { get; set; }
        public string Reason { get; set; } = string.Empty;
        public LeaveRequestStatus Status { get; set; } = LeaveRequestStatus.Pending;
        
    }
}
